src/containers/Accounts/AccountNFTTable/AccountNFTTable.tsx

```diff
@@ -1,85 +1,82 @@
-import React, { useContext, useEffect, useState } from 'react'
+import React, { useContext } from 'react'
 import { useTranslation } from 'react-i18next'
-import { connect } from 'react-redux'
-import { bindActionCreators } from 'redux'
-import { useParams } from 'react-router'
-import { XrplClient } from 'xrpl-client'
+import { useInfiniteQuery } from 'react-query'
+import { Link } from 'react-router-dom'
 
-import { concatNFT } from '../../shared/utils'
-import { loadAccountNFTs } from './actions'
+import { ANALYTIC_TYPES, analytics } from '../../shared/utils'
 import Loader from '../../shared/components/Loader'
 import SocketContext from '../../shared/SocketContext'
-import { AccountNFToken } from '../../shared/transactionUtils'
-import { AccountNFTState } from './reducer'
+import { EmptyMessageTableRow } from '../../shared/EmptyMessageTableRow'
+import { getAccountNFTs } from '../../../rippled/lib/rippled'
+import Account from '../../shared/components/Account'
 
-interface Props {
-  loading: boolean
-  data: AccountNFTState['data']
-  actions: {
-    loadAccountNFTs: (
-      accountId: string,
-      marker: string | undefined,
-      socket: XrplClient | undefined,
-    ) => {}
-  }
+export interface AccountNFTTableProps {
+  accountId: string
 }
 
-const AccountNFTTableDisconnected = (props: Props) => {
-  const [nfts, setNfts] = useState<AccountNFToken[]>([])
-  const [marker, setMarker] = useState<string | undefined>(undefined)
-  const { t } = useTranslation()
-  const { id: accountId } = useParams<{ id: string }>()
-  const { actions, data, loading } = props
+export const AccountNFTTable = ({ accountId }: AccountNFTTableProps) => {
   const rippledSocket = useContext(SocketContext)
-
-  useEffect(() => {
-    if (data.nfts === undefined) return
-    setMarker(data.marker)
-    setNfts((oldNfts: AccountNFToken[]): AccountNFToken[] =>
-      concatNFT(oldNfts, data.nfts || []),
-    )
-  }, [accountId, data])
-
-  useEffect(() => {
-    setNfts([])
-    setMarker(undefined)
-    actions.loadAccountNFTs(accountId, undefined, rippledSocket)
-  }, [accountId])
-
-  const loadMoreNfts = () => {
-    actions.loadAccountNFTs(accountId, marker, rippledSocket)
-  }
+  const {
+    data: pages,
+    isFetching: loading,
+    hasNextPage,
+    fetchNextPage,
+  } = useInfiniteQuery(
+    ['account_nfts', accountId],
+    async ({ pageParam = '' }) =>
+      getAccountNFTs(rippledSocket, accountId, pageParam).catch(
+        (errorResponse) => {
+          const errorLocation = `account NFTs ${accountId} at ${pageParam}`
+          analytics(ANALYTIC_TYPES.exception, {
```

Comment:
Just curious, we are not handling any errors here right? We would just show an empty table even if there is error thrown?

Reply:
I will look into this.

Reply:
It shows no nfts found.

![Screen Shot 2022-08-29 at 5 37 01 PM](https://user-images.githubusercontent.com/464895/187311016-74ad4ba7-36ec-4808-ae67-62869d6b0fec.png)

Reply:
I will create a ticket to handle exceptions better in this component.  I have not found a consistent way this is handled.

Reply:
Was just curious, I knew it would show the empty message. Just making sure its okay to ignore errors here 😁